lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -279,7 +284,7 @@ struct StructuralHasher {
       // Skip names and annotations, except in certain cases.
```

Comment:
Stray comment while looking:

```
      value.walk([&](FlatSymbolRefAttr) { hasSeenSymbol = true; });
```

This attribute walking constructs a fresh AttrTypewalker which uses a DenseMap to cache the Attribute -> WalkResult which can be slow if done every time instead of reusing a walker created once.

The re-use might save re-walking the same attributes in order to determine if they have a symbol?

Anyway something to keep in mind/consider, not sure it's problematic as-is and forgive the comment unrelated to this PR.